common/flags/ClassScanner.java

```diff
@@ -128,8 +137,28 @@ private JarFile getJarFile(URL url) throws IOException {
     return result.build();
   }
 
+  public void scanClass(Class clazz, Map<String,FlagData> flags) {
```

Comment:
Since most of the code is the same, it'd be good to try and share code with `scanPackage()` instead of duplicate.